lib/models/itinerary.dart

```diff
@@ -1,54 +1,97 @@
-      //JSON contains all recommended routes, any transportation mode changes, and how many steps in the directions.
-      _length = _parsedJson["routes"][0]["legs"][0]["steps"].length;
-      print(_length);
-      //safety prevents the JSON from being parsed multiple times unnecessarily
-      if (safety) {
-        for (int i = 0; i < _length; i++) {
-          String s1 = _parsedJson["routes"][0]["legs"][0]["steps"][i]
-                  ["html_instructions"]
-              .replaceAll(
-                  //Regex to replace certain special characters in HTML with whitespace
-                  RegExp(r'(<\/?\w+\/?>?| \w+=\"\w+-\w+:\d.\d\w+\">)'),
-                  ' ');
-          String s2 = _parsedJson["routes"][0]["legs"][0]["steps"][i]
-              ["duration"]["text"];
-          String s3 = _parsedJson["routes"][0]["legs"][0]["steps"][i]
-              ["distance"]["text"];
-          _itinerary.addAll({
-            s1: {s2: s3}
-          });
-        }
-        safety = false;
-      }
+    //JSON contains all recommended routes, any transportation mode changes, and how many steps in the directions.
+    Map<String, Map<String, String>> temp = Map<String, Map<String, String>>();
+    var steps = rawJson["routes"][0]["legs"][0]["steps"].length;
+    for (int i = 0; i < steps; i++) {
+      String s1 = rawJson["routes"][0]["legs"][0]["steps"][i]
+              ["html_instructions"]
+          .replaceAll(
+              //Regex to replace certain special characters in HTML with whitespace
+              RegExp(r'(<\/?\w+\/?>?| \w+=\"\w+-\w+:\d.\d\w+\">)'),
+              ' ');
+      String s2 =
+          rawJson["routes"][0]["legs"][0]["steps"][i]["duration"]["text"];
+      String s3 =
+          rawJson["routes"][0]["legs"][0]["steps"][i]["distance"]["text"];
+      temp.addAll({
+        s1: {s2: s3}
+      });
+    }
+    return temp;
+  }
+
+  static List<Polyline> getPolylinePoints(Map<String, dynamic> rawJson) {
+    PolylinePoints tPolylinePoints = PolylinePoints();
+    List<PointLatLng> tPointLatLng = [];
+    List<Polyline> tPolyline = [];
+
+    double sLat = rawJson['routes'][0]['legs'][0]['start_location']['lat'];
+    double sLong = rawJson['routes'][0]['legs'][0]['start_location']['long'];
+    double eLat = rawJson['routes'][0]['legs'][0]['end_location']['lat'];
+    double eLong = rawJson['routes'][0]['legs'][0]['end_location']['long'];
+
+    String sDesc = rawJson['routes'][0]['legs'][0]['start_address'];
+    String eDesc = rawJson['routes'][0]['legs'][0]['end_address'];
+
+    var steps = rawJson['routes'][0]['legs'][0]['steps'];
+
+    for (int i = 0; i < steps.length; i++) {
+      dynamic directions = steps[i]["polyline"]["points"];
+      List<PointLatLng> temp = tPolylinePoints.decodePolyline(directions);
+      tPointLatLng = new List.from(tPointLatLng)..addAll(temp);
+    }
+    List<LatLng> po = [];
+    tPointLatLng.forEach((f) {
+      po.add(LatLng(f.latitude, f.longitude));
```

Comment:
This is a bit hard to understand what it's doing at first glance.
Maybe add some comments for blocks of code

Reply:
Also you can remove those 4 double variables at the top if theyre not used

Reply:
I'm gonna keep the ones for involving starting location info i.e. `sLat, sLong, and sDesc` for now. They might be used in when viewing the list of instructions in (`/lib/widgets/directions_drawer.dart`)

Reply:
fixed in 29ee4c271ab5830164da28b1f0d65a239c921d08